Dockerfile

```diff
@@ -96,30 +95,38 @@ FROM acousticbrainz-base AS acousticbrainz-prod
 
 ARG deploy_env
 
-RUN pip install uWSGI==2.0.17.1
+RUN pip install --no-cache-dir uWSGI==2.0.17.1
+
+RUN mkdir /cache_namespaces
+RUN chown -R www-data:www-data /cache_namespaces
 
 # Consul template service is already set up, just need to copy the configuration
 COPY ./docker/consul-template.conf /etc/consul-template.conf
 
+# runit service files
+# All services are created with a `down` file, preventing them from starting
+# rc.local removes the down file for the specific service we want to run in a container
+# http://smarden.org/runit/runsv.8.html
+
 # uwsgi service files
-COPY ./docker/$deploy_env/uwsgi/uwsgi.service /etc/service/uwsgi/run
-RUN chmod 755 /etc/service/uwsgi/run
-COPY ./docker/$deploy_env/uwsgi/uwsgi.ini /etc/uwsgi/uwsgi.ini
+COPY ./docker/uwsgi/uwsgi.service /etc/service/uwsgi/run
+COPY ./docker/uwsgi/uwsgi.ini /etc/uwsgi/uwsgi.ini
+RUN touch /etc/service/uwsgi/down
 
-RUN mkdir /cache_namespaces
-RUN chown -R www-data:www-data /cache_namespaces
+# hl_extractor service files
+COPY ./docker/hl_extractor/hl_extractor.service /etc/service/hl_extractor/run
+RUN touch /etc/service/hl_extractor/down
 
-# Create a user named acousticbrainz for cron jobs
-RUN useradd --create-home --shell /bin/bash acousticbrainz
+# dataset evaluator service files
+COPY ./docker/dataset_eval/dataset_eval.service /etc/service/dataset_eval/run
+RUN touch /etc/service/dataset_eval/down
 
 # Add cron jobs
-ADD docker/crontab /etc/cron.d/ab-crontab
-RUN chmod 0644 /etc/cron.d/ab-crontab && crontab -u acousticbrainz /etc/cron.d/ab-crontab
-RUN touch /var/log/stats_calc.log /var/log/stats_cache.log && chown acousticbrainz:acousticbrainz /var/log/stats_calc.log /var/log/stats_cache.log
-
-# Make sure that the cron service doesn't start automagically
-# http://smarden.org/runit/runsv.8.html
+ADD docker/crontab /etc/cron.d/acousticbrainz
+RUN chmod 0644 /etc/cron.d/acousticbrainz
 RUN touch /etc/service/cron/down
 
+COPY ./docker/$deploy_env/rc.local /etc/rc.local
```

Comment:
Instead of duplicating scripts, what about using a common one and env var passed to the container ?
https://github.com/phusion/baseimage-docker#environment-variables

Reply:
Currently (in listenbrainz), we have separate container names for production and beta:
https://github.com/metabrainz/listenbrainz-server/blob/6cd87707144cac9bafa2529f9c93e3c3f48ff5cc/docker/prod/uwsgi/uwsgi.service
https://github.com/metabrainz/listenbrainz-server/blob/6cd87707144cac9bafa2529f9c93e3c3f48ff5cc/docker/beta/uwsgi/uwsgi.service

We could definitely have a single rc.local file that shares the same container names in beta and prod, I'm not sure what this implies in terms of startup scripts/consul, and I don't know if it has the potential to cause confusion when we're trying to debug a problem under pressure where beta/prod are getting mixed up for some reason...

Reply:
I think keeping it like this for now is the better idea because this PR has been lingering for a long while.

[AB-408](https://tickets.metabrainz.org/browse/AB-408)